packages/dds/tree/src/test/shared-tree-core/utils.ts

```diff
@@ -86,4 +92,38 @@ export class TestSharedTreeCore extends SharedTreeCore<DefaultEditBuilder, Defau
 	public override getLocalBranch(): SharedTreeBranch<DefaultEditBuilder, DefaultChangeset> {
 		return super.getLocalBranch();
 	}
+
+	protected override submitCommit(
+		...args: Parameters<SharedTreeCore<DefaultEditBuilder, DefaultChangeset>["submitCommit"]>
+	): void {
+		// We do not submit ops for changes that are part of a transaction.
+		if (this.transactionStart === undefined) {
+			super.submitCommit(...args);
+		}
+	}
+
+	public startTransaction(): void {
```

Comment:
@yann-achard-MS I had to add "transaction" capability to SharedTreCoreTest in order to preserve the ability to unit test the resubmitMachine during a transaction. I'm not sure how else to do it without some serious refactoring. This isn't that bad, but I'm hoping that in future refactors, we'll be able to untangle it even more. I'm thinking that maybe we can pull a "TransactionManager" or something out of TreeCheckout and let it be instantiated in isolation.